src/core_modules/capture-core/utils/url.js

```diff
@@ -10,3 +10,21 @@ export const urlArguments = (programId: string, orgUnitId: string): string => {
 
     return argArray.join('&');
 };
+
+type Url = {programId?: string, orgUnitId?: string, trackedEntityTypeId?: string}
+
+export const urlThreeArguments = ({ programId, orgUnitId, trackedEntityTypeId }: Url): string => {
+    const argArray = [];
+
+    if (orgUnitId) {
+        argArray.push(`orgUnitId=${orgUnitId}`);
+    }
+
+    if (programId) {
+        argArray.push(`programId=${programId}`);
+    } else if (trackedEntityTypeId) {
+        argArray.push(`trackedEntityTypeId=${trackedEntityTypeId}`);
+    }
+
+    return argArray.join('&');
+};
```

Comment:
I wanted to have your opinion on this approach before I go on and replace the function above everywhere in the app. If you think that the implementation described here can work I can go on and do the replacement on the next PR.